github-contribution-guideline.md

```diff
@@ -0,0 +1,122 @@
+
+## Discuss
+
+It proved to be useful to discuss about an articles content before writing the article. Ask your colleagues, maybe they have some valuable input! 
+
+   * Use the [Issues on Github](https://github.com/Zuehlke/zuehlke-book/issues): discuss directly in the issue you created - and don't miss to give feedback to other announced articles.
+
+   * Use Yammer, a coffee break, a team event or similar
+
+## Write 
+
+**Content Conventions:**
+
+* written in English
+
+* short enough to be read in 10 minutes: size should be less than 2 A4 pages long in printout PDF version (that is between 400 and 750 words)
+
+* reflects on a single subject only (better make several articles, if you have too much material / ideas to write about)
+
+* matches our [Vision](./README.md#vision): it is about practicaly proven concepts, best practices, cultural philosophies or great ideas that we use in our daily work at Zühlke.
+
+* the subject is related to our work and the advice concrete enough to consider.
+
+* the subject is something you have experience on in some real projects and not just theoretical ideas
+
+**Language Conventions:**
+
+@Rolf: Please add title guideline
+
+**Work directly with GitHub:**
+
+We are using this Git repository, so contributing an article is as easy as contributing source code in any project. Every article is written in your personal fork of the repository as a simple markdown (*.md) file and reviewed and published by using Pull Requests. 
+
+If this sounds complicated to you, just carefully follow these very easy steps to create a new article directly on the github web page:
+
+  1. Use this link to [fork the Zuehlke/zuehlke-book repository](https://github.com/Zuehlke/zuehlke-book/fork).
+
+  2. Navigate to the `articles` folder in your forked repository. The URL looks something like:
+  https://github.com/{username}/zuehlke-book/tree/develop/articles
+
+  3. Enter a unique name for your article file followed by the *.md file extension. Example: `my-funny-article.md` (can be changed later)
+
+  4. Copy the content of [Article Example as template to start with](https://raw.githubusercontent.com/Zuehlke/zuehlke-book/develop/articles/_example.md) into the file
+    
+  5. Adjust `authorName` and `authorGithubUsername` at the top of the file
+    
+  6. Commit the new file
+
+Congratulations, you have created the article file in your repository! You can now continue to work on your article incrementally directly in GitHub. If you prefer your own text editor: feel free to use your own "clone-edit-commit-push"- workflow with your local git client.
+
+Please apply the following formal rules:
+
+  * Use [GitHub Flavored Markdown Syntax](https://guides.github.com/features/mastering-markdown/) in the markdown file.
+
+  * In all your commits you should use the issue number of your article prefixed with `#` .  
+   Example: `#15 add article to Table of Content`.
```

Comment:
I do not see a benefit in adding the issue number. @bruderol agree to let authors decide for themselfes about the commit comment?

Reply:
There is none: We either have commits on the file source of the article or the commits are grouped together in a pull request. Associate the pull request with the issue number and we have a full trace.

Reply:
it says you "should" use. Nobody will be killed if he forgets it. Nevertheless I think it is good practice to do so, otherwise you can not see for which article or issue a commit was for in the history. But ok, we can drop it, if you prefer and think this is so complicated.

Reply:
reformulated to "It is good practice to. ..."